src/routing/mod.rs

```diff
@@ -312,17 +311,42 @@ impl Routing {
         dst: DstLocation,
         content: Bytes,
     ) -> Result<()> {
+        if let DstLocation::EndUser(EndUser::Client {
+            socket_id,
+            public_key,
+        }) = dst
+        {
+            let mut socket = None;
```

Comment:
This can be written more succinctly:

```rust
let socket = self.stage.state.lock().await.get_socket_addr(&socket_id).copied();
if let Some(socket) = socket {
    return self.send_message_to_client(...);
} else {
   ...
}
```